server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -710,9 +714,27 @@ static boolean shouldPreFilterSearchShards(ClusterState clusterState,
         } else if (preFilterShardSize == null) {
             preFilterShardSize = SearchRequest.DEFAULT_PRE_FILTER_SHARD_SIZE;
         }
+        if (RollupV2.isEnabled() && hasRollupDatastream(indices, searchRequest.indices(), clusterState)) {
```

Comment:
I'd prefer that we always run the can match phase if a datastream is present. No need to check the existence of rollups, the can match phase was made to handle datastreams before they existed ;).